src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj

```diff
@@ -32,6 +32,7 @@
     <PackageReference Include="Microsoft.VisualStudio.Imaging" Version="$(MicrosoftVisualStudioImagingVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Language.StandardClassification" Version="$(MicrosoftVisualStudioLanguageStandardClassificationVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Language.Intellisense" Version="$(MicrosoftVisualStudioLanguageIntellisenseVersion)" />
+    <PackageReference Include="Microsoft.VisualStudio.RemoteControl" Version="$(MicrosoftVisualStudioRemoteControlVersion)" />
```

Comment:
This may be problematic, since this now means VS for Mac needs to ship this binary if they're not. Do we need to move this _up_ a layer to the VS layer rather than down?

Reply:
This is a small, netstandard2.0-compatible binary. Today it's used by our Remote.ServiceHub implementation; perhaps we could reorganize the code to remove this requirement but if it doesn't cause a problem it might not be worth the hurdles.

Reply:
We already ship RemoteControl. Here's the list of all assemblies in VSMac for reference: https://gist.github.com/KirillOsenkov/7d829534a037c2b3545c95d569f34469

Reply:
@KirillOsenkov: even if you're shipping it, does it function? It looks like the code we see uses a bunch of registry and Windows-specific APIs -- do you have a cross-plat version of it or something else?

Reply:
Registry is implemented under Mono. The PInvokes are only called on Windows, under guarded runtime checks. I've checked the code under the `VSRemoteControl` repo, and it all looks okay.

Reply:
Since we confirmed that this is fine for VSMac, can we take this change as is now? @jasonmalinowski

Reply:
There's still the problem that this PR is also moving some DllImports to windows-only APIs as well:

https://github.com/dotnet/roslyn/blob/2998dd44640da29d600d93b2bb6764a804a0d4be/src/EditorFeatures/Core/SymbolSearch/Patching/Delta.cs#L54

So we at least need to confirm that won't be causing issues either.

@genlu: is this blocking something?

Reply:
Yea, until we have our remote targeting .Net Core, we can't proceed to get those bits in our package and test roslyn servicehub running on CoreClr.

Reply:
How about adding a check to the feature that uses this and disabling it when we are not running on Windows?

Reply:
Can refactor this to use MEF and keep the implementation of SymbolSerchEngine in EditorFeature.Wpf?

Reply:
We should definitely move the SymbolSerchEngine to EditorFeatures. The only thing that needs to be platform-conditional is the the feature that calls these P/Invokes. My understanding is that the affected feature is API lookup in non-referenced nuget packages. On non-Windows this feature can return no symbols.

Reply:
I'm not so sure about that, it looks the data for referenced assemblies are stored in the same database used by "find package", which probably means it share the same patching mechanism as mentioned by @jasonmalinowski above. I might be wrong though, @CyrusNajmabadi could you please confirm if it is the case?

Reply:
>We should definitely move the SymbolSerchEngine to EditorFeatures. The only thing that needs to be platform-conditional is the the feature that calls these P/Invokes. 

Would something like this work?
https://github.com/sharwell/roslyn/pull/2

Reply:
er.  i'm not exactly sure what is being asked here.  however:

1. i don't really see a problem with moving this code down.
2. the existing code does make use of windows-specific functionality.  specifically it pinvokes into msdelta.dll.  This just won't work on mac.
3. both 'nuget search' and 'reference assembly' search go through this, so neither will work.
4.  all thsi work is kicked off from VisualStudioSymbolSearchService which is in the VS.LanguageServices dll.  So i don't think it has any impact on VS4Mac, or anyone else.

Given that this effectively will not be running on vs4mac, i don't see any problems here and i think it's fine to have this moved down.

Reply:
Hmm, if VisualStudioSymbolSearchService kicks all this off then I'm not sure why we rolled back my PR that did this move.

Reply:
If it was to make sure then I think we can just revert the revert, now that we know.

Reply:
@tmat: rolling back was easier than confirming it was safe given it landed right before a critical snap. Had it merged a few days earlier then we could have just had the conversation we're having here and sorted it out in time.

More generally: EditorFeatures is for code that does not have a platform specific dependencies; EditorFeatures.Wpf is allowed to take Windows-specific dependencies. If this is only being used by VS code, why can't we just move the stuff doing the P/Invokes to the VS layer?

Reply:
> If this is only being used by VS code, why can't we just move the stuff doing the P/Invokes to the VS layer?

The implementation doesn't depend on VS and should run in ServiceHub process, which does not depend on VS.

Reply:
>why can't we just move the stuff doing the P/Invokes to the VS layer?

We want this feature to be available when our remote is targeting .NetCore and run on Windows machine, but also want to eliminate dependency on VS/Windows from the remote project (so we can actually target it to .NetCore).

Reply:
A conditional dependency on Windows is ok. The compiler also has a conditional dependency on Windows (`/debug:full`).
Dependency on VS is not ok.

Reply:
@tmat So is this good to merge as-is?

Reply:
I think so, as long as we verified it indeed does not try to invoke the Windows-specific code path on Mac.

Reply:
It doesn't add anything to the MEF composition, so the only way this could get called on Mac is if the code is being invoked explicitly.

Reply:
If we want to make sure, we can take this change to guard against potential misuse in the future.
https://github.com/sharwell/roslyn/pull/2

Reply:
@genlu Can you submit that as a follow-up pull request?

Reply:
Yea, sure. Will do if we think it'd help.

Reply:
https://github.com/dotnet/roslyn/pull/45764